### PR TITLE
Base for Biosample Sample split

### PR DESCRIPTION
**Pull request by jleblond**

### CODE REVIEW

File: backend/fms_core/migrations/0026_v3_5_0.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                ('created_by', models.ForeignKey(blank=True, on_delete=django.db.models.deletion.PROTECT,
+                                                 related_name='fms_core_derivedsample_creation',
+                                                 to=settings.AUTH_USER_MODEL)),
+                ('sample_kind', models.ForeignKey(
+                    help_text='Biological material collected from study subject during the conduct of a genomic study project.',
+                    on_delete=django.db.models.deletion.PROTECT, to='fms_core.samplekind')),
+                ('updated_by', models.ForeignKey(blank=True, on_delete=django.db.models.deletion.PROTECT,
+                                                 related_name='fms_core_derivedsample_modification',
+                                                 to=settings.AUTH_USER_MODEL)),
+                ('sample', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.PROTECT,
```
Comment (sebastianballesteros) [inline backend/fms_core/migrations/0026_v3_5_0.py:85]: Did we say 'samples' was ManyToMany or it should stay as a ForeignKey 'sample'?

Comment (UlysseFG) [reply backend/fms_core/migrations/0026_v3_5_0.py:85]: I think this is a temporary field used during the migration ...

File: backend/fms_core/migrations/0026_v3_5_0.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+                -- Create DerivedBySample based on parent-only samples based on SampleLineage
+                INSERT INTO fms_core_derivedbysample (sample_id, derived_sample_id, volume_ratio, created_at, created_by_id, updated_at, updated_by_id, deleted)
+                SELECT DISTINCT(samplelineage.parent_id), derivedsample.id, 1, sample.created_at, sample.created_by_id,
+                       current_timestamp, 1, FALSE
+                FROM fms_core_samplelineage samplelineage
+                JOIN fms_core_sample sample ON sample.id = samplelineage.parent_id
+                JOIN fms_core_derivedsample derivedsample ON derivedsample.sample_id = parent_id
+                JOIN fms_core_processmeasurement processmeasurement ON processmeasurement.id = samplelineage.process_measurement_id
+                WHERE parent_id NOT IN (SELECT child_id FROM fms_core_samplelineage);
+            """
```
Comment (sebastianballesteros) [inline backend/fms_core/migrations/0026_v3_5_0.py:237]: Might be a good idea to add migrations.RunSQL.noop to be able to rollback to a previous migration if needed.

File: backend/fms_core/migrations/0026_v3_5_0.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                           derivedsamplelineage.source_id AS sample_source_id,
+                           derivedsample.id AS derivedsample_id,
+                           derivedsamplelineage.original AS is_original_derivedsample,
+                           derivedsamplelineage.level AS level,
+                           derivedsamplelineage.pm_created_at,
+                           derivedsamplelineage.pm_created_by_id
+                    FROM derivedsamplelineage
+                    JOIN fms_core_derivedsample derivedsample ON derivedsample.sample_id = derivedsamplelineage.source_id
+                ) q;
+            """
```
Comment (sebastianballesteros) [inline backend/fms_core/migrations/0026_v3_5_0.py:295]: Same comment as last query

**Pull request merged**